cpp/src/cylon/table.cpp

```diff
@@ -1756,5 +1761,118 @@ Status WriteParquet(const std::shared_ptr<cylon::CylonContext> &ctx_,
 
   return Status(Code::OK);
 }
+
+/**
+ * Slice the part of table to create a single table
+ * @param table, offset and length
+ * @return new sliced table
+ */
+
+
+Status Slice(const std::shared_ptr<Table> &in, int64_t offset, int64_t length,
+              std::shared_ptr<cylon::Table> &out) {
+#ifdef CYLON_DEBUG
+  auto p1 = std::chrono::high_resolution_clock::now();
+#endif
+  const auto &ctx = in->GetContext();
+  auto pool = cylon::ToArrowPool(ctx);
+  std::shared_ptr<arrow::Table> out_table, in_table = in->get_table();
+  std::vector<int> cols = {0};
+  Unique(in, cols, out, true);
+
+
+  if (!in->Empty()) {
+    if (in_table->column(0)->num_chunks() > 1) {
+      CYLON_ASSIGN_OR_RAISE(in_table, in_table->CombineChunks(pool))
+    }
+
+    std::unique_ptr<TableRowIndexEqualTo> row_comp;
+    RETURN_CYLON_STATUS_IF_FAILED(TableRowIndexEqualTo::Make(in_table, cols, &row_comp));
+
+    std::unique_ptr<TableRowIndexHash> row_hash;
+    RETURN_CYLON_STATUS_IF_FAILED(TableRowIndexHash::Make(in_table, cols, &row_hash));
+
+    const int64_t num_rows = length;
+    ska::bytell_hash_set<int64_t, TableRowIndexHash, TableRowIndexEqualTo>
+        rows_set(num_rows, *row_hash, *row_comp);
+
+    arrow::Int64Builder filter(pool);
+    RETURN_CYLON_STATUS_IF_ARROW_FAILED(filter.Reserve(num_rows));
+#ifdef CYLON_DEBUG
+    auto p2 = std::chrono::high_resolution_clock::now();
+#endif
+    for (int64_t row = offset; row < offset + num_rows; ++row) {
+      const auto &res = rows_set.insert(row);
+      if (res.second) {
+        filter.UnsafeAppend(row);
+      }
+    }
+#ifdef CYLON_DEBUG
+    auto p3 = std::chrono::high_resolution_clock::now();
+#endif
+    rows_set.clear();
+#ifdef CYLON_DEBUG
+    auto p4 = std::chrono::high_resolution_clock::now();
+#endif
+    CYLON_ASSIGN_OR_RAISE(auto take_arr, filter.Finish());
+    CYLON_ASSIGN_OR_RAISE(auto take_res, arrow::compute::Take(in_table, take_arr))
+    out_table = take_res.table();
+
+#ifdef CYLON_DEBUG
+    auto p5 = std::chrono::high_resolution_clock::now();
+    LOG(INFO) << "P1 " << std::chrono::duration_cast<std::chrono::milliseconds>(p2 - p1).count()
+              << " P2 " << std::chrono::duration_cast<std::chrono::milliseconds>(p3 - p2).count()
+              << " P3 " << std::chrono::duration_cast<std::chrono::milliseconds>(p4 - p3).count()
+              << " P4 " << std::chrono::duration_cast<std::chrono::milliseconds>(p5 - p4).count()
+              << " tot " << std::chrono::duration_cast<std::chrono::milliseconds>(p5 - p1).count()
+              << " tot " << rows_set.load_factor() << " " << rows_set.bucket_count();
+#endif
+  } else {
+    out_table = in_table;
+  }
```

Comment:
You don't need to do this for local slicing. You can simply call `arrow::Table::Slice`
https://github.com/apache/arrow/blob/389587c566e0d0d59b635a76fcc8dbb89358d6ec/cpp/src/arrow/table.h#L118

I'm curious as to why we need to build a hashset for slice indices. :thinking:

Reply:
For the Slice operation, we don't need to perform hash. I added this for future operations. For example, in the case of slicing operation by removing duplicate entries. But that was not added though. 
I did not know the  API for arrow::Table::Slice there. I will make a change for that.

Reply:
I dont think removing duplicates is required for slicing, and since we already have that capability, IMO it would duplicate the functionality.